ddl/dm_schema_tracker.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl
+
+import (
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/table/tables"
+)
+
+// infoStore is a simple structure that stores DBInfo and TableInfo. It's not thread-safe.
+type infoStore struct {
+	lowerCaseTableNames int // same as variable lower_case_table_names
```

Comment:
It seems that this is not needed at present. Can you remove it?

Reply:
it's needed in `ciStr2Key`. In DM's use case, upstream is MySQL so we should set lowerCaseTableNames from it.

Reply:
But I see `lowerCaseTableNames` is read-only.

Reply:
Sorry, in the big PR https://github.com/pingcap/tidb/pull/35867, I use struct literal to build infoStore in this package. I can write a newInfoStore.

Reply:
Got it. DM needs use it